pkg/operation/botanist/controlplane.go

```diff
@@ -996,11 +996,15 @@ func (b *Botanist) DeployKubeControllerManager() error {
 		},
 	}
 
-	replicaCount, err := common.CurrentReplicaCount(b.K8sSeedClient.Client(), b.Shoot.SeedNamespace, v1beta1constants.DeploymentNameKubeControllerManager)
-	if err != nil {
-		return err
+	if b.Shoot.HibernationEnabled == b.Shoot.Info.Status.IsHibernated {
+		// Keep the replica count same if the shoot is not transitioning from/to hibernation state
+		// otherwise this may interfere with dependency-watchdog
+		replicaCount, err := common.CurrentReplicaCount(b.K8sSeedClient.Client(), b.Shoot.SeedNamespace, v1beta1constants.DeploymentNameKubeControllerManager)
+		if err != nil {
+			return err
+		}
+		defaultValues["replicas"] = replicaCount
 	}
```

Comment:
The defaultValues["replicas"] from the chart contains 1 - so this is now actually scaling up the KCM if the Shoot is not hibernated.  Can we make this clear in a comment somewhere for code better readability?

Reply:
WDYM @danielfoehrKn ? If the shoot is not hibernated then it should be deployed with replicas=1, similar to every other component we deploy. Why do we need a comment for this? Or am I overseeing something?

Reply:
OK, as discussed, we should not use any Helm values as defaults. Instead, we should always explicitly inject the value as we need them for the various charts. The `values.yaml` file should only be an example showing which values are possible in general.
We should straighten this soon. For now, let's continue with this PR.